### ISSUE TITLE
Editor Output window regression, RichTextLabel does not always update its scrollbar properly

### ISSUE THREAD

**Issue by bitsawer** [godotengine/godot#78434]
### Godot version

4.1.dev 116f783db73f4bf7e9e96ae54dd3d0a20337cc8a

### System information

Godot v4.1.beta (116f783db) - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 970 (NVIDIA; 31.0.15.1694) - Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz (8 Threads)

### Issue description

Editor Output window RichTextLabel does not always update its scrollbar properly after print() calls. Looks like a regression from https://github.com/godotengine/godot/pull/78241, reverting it fixes this issue (but of course restores the bug it was fixing). CC @bruvzg 

### Steps to reproduce

1) Open the demo project
2) Make the Output window visible in the bottom view, press "Clear Output"
2) Run the project, notice how the output scroll does not follow the actual content added. In fact, the vertical scroll bar is not even visible. However, manually resizing the bottom view updates the layout and the scrollbar becomes visible and "jumps" to the end as expected.

Looks like printing a single line in _process() doesn't always update the scrolling. However, uncommenting the line in _ready() to print many lines in a single frame seems to be enough to update the scrolling and it starts working for the _process() prints, too.

### Minimal reproduction project

[OutputScrollBug.zip](https://github.com/godotengine/godot/files/11786216/OutputScrollBug.zip)

**Comment by YuriSizov** [comment]
At this point fixing it for 4.1 is likely very risky, as any change in the sizing logic can cause unforeseeable side effects and we don't have much time to test it before going stable. We should fix it for 4.2, and if the fix proves to be stable, we can cherry-pick it.

**Comment by bruvzg** [comment]
> At this point fixing it for 4.1 is likely very risky, as any change in the sizing logic can cause unforeseeable side effects and we don't have much time to test it before going stable.

If it's a sizing issue, yes. But it's likely a multithreaded processing issue.

I can't test in right now, but try applying the following patch:
```diff
@@ -2908,6 +2908,8 @@ void RichTextLabel::_process_line_caches() {
 		int sr = MIN(main->first_invalid_font_line.load(), main->first_resized_line.load());
 		main->first_invalid_font_line.store(fi);
 
+		if (sr != 0) {
+			total_height = _calculate_line_vertical_offset(main->lines[sr - 1]);
+		}
 		for (int i = sr; i < fi; i++) {
 			total_height = _resize_line(main, i, theme_cache.normal_font, theme_cache.normal_font_size, text_rect.get_size().width - scroll_w, total_height);
 			total_height = _update_scroll_exceeds(total_height, ctrl_height, text_rect.get_size().width - scroll_w, i, old_scroll, text_rect.size.height);
```

**Comment by YuriSizov** [comment]
@Sauermann Feel free to open a PR!

**Comment by Sauermann** [comment]
I would consider that only after understanding the patch in detail.

**Comment by YuriSizov** [comment]
I can't reproduce the bug myself, but I do notice a difference in behavior with this change applied. Without this patch, the scroll only appears after all the lines have been printed, so basically the calculation is delayed. With this patch, the scroll is immediately updated as more text is printed.

I think as a result the log updates a slightly laggier (based solely on empirical evidence). Though it's still fast enough on my PC.

**Comment by akien-mga** [comment]
Tentatively putting this back on the 4.1 milestone as several users mentioned it to be quite annoying. bruvzg will only have time to test a potential fix further from Thursday, so we can't guarantee that it will make it in 4.1, but let's keep it tracked for now.

**Comment by AThousandShips** [comment]
Seems to not be fully solved #78899

**Comment by bibizu** [comment]
I am still having a similar issue pertaining to this in v4.1.1, but it appears not to be the one described here but more similar to #79975, in that my scroll bar does still appear, but it simply does not follow output. The workarounds mentioned here don't seem to work for me either.

**Comment by Sauermann** [comment]
@bibizu since you experience a similar but not the same problem, I would ask you to create a new issue for that bug.
